cfme/tests/cloud/test_instance_ownership.py

```diff
@@ -0,0 +1,157 @@
+    login.login_admin()
+    user2 = new_user(group_only_user_owned)
+    yield user2
+    login.login_admin()
+    user2.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def user3(group_user_or_group_owned):
+    login.login_admin()
+    user3 = new_user(group_user_or_group_owned)
+    yield user3
+    login.login_admin()
+    user3.delete()
+
+
+def new_user(group_only_user_owned):
+    user = ac.User(name='user_' + fauxfactory.gen_alphanumeric(),
+                credential=new_credential(),
+                email='abc@redhat.com',
+                group=group_only_user_owned,
+                cost_center='Workload',
+                value_assign='Database')
+    user.create()
+    return user
+
+
+def test_form_button_validation(request, user1, setup_cloud_provider):
+    set_instance_to_user = EC2Instance('cu-9-5', setup_cloud_provider)
+    # Reset button test
+    set_instance_to_user.set_ownership(user=user1.name, click_reset=True)
+    # Cancel button test
+    set_instance_to_user.set_ownership(user=user1.name, click_cancel=True)
+    # Save button test
+    set_instance_to_user.set_ownership(user=user1.name)
+    # Unset the ownership
+    set_instance_to_user.unset_ownership()
+
+
+def test_user_ownership_crud(request, user1, setup_cloud_provider):
+    set_instance_to_user = EC2Instance('cu-9-5', setup_cloud_provider)
+    # Setting the ownership and checking it
+    set_instance_to_user.set_ownership(user=user1.name)
+    # pdb.set_trace()
+    login.login(user1.credential.principal, user1.credential.secret)
+    assert(set_instance_to_user.does_vm_exist_in_cfme(), "Instance not found")
+    login.login_admin()
+    # Unset the ownership
+    set_instance_to_user.unset_ownership()
+    assert(not set_instance_to_user.does_vm_exist_in_cfme(), "Instance exists")
+
+
+def test_group_ownership_on_user_only_role(request, user2, setup_cloud_provider):
+    set_instance_to_group = EC2Instance('cu-9-5', setup_cloud_provider)
+    # pdb.set_trace()
+    set_instance_to_group.set_ownership(group=user2.group.description)
+    login.login(user2.credential.principal, user2.credential.secret)
+    assert(set_instance_to_group.does_vm_exist_in_cfme(), "Instance not found")
+    login.login_admin()
+    # Unset the ownership
+    set_instance_to_group.unset_ownership()
+
+
+def test_group_ownership_on_user_or_group_role(request, user3, setup_cloud_provider):
+    set_instance_to_group = EC2Instance('cu-9-5', setup_cloud_provider)
+    set_instance_to_group.set_ownership(group=user3.group.description)
+    login.login(user3.credential.principal, user3.credential.secret)
+    assert(set_instance_to_group.does_vm_exist_in_cfme(), "Instance not found")
+    login.login_admin()
+    # Unset the ownership
+    set_instance_to_group.unset_ownership()
+    login.login(user3.credential.principal, user3.credential.secret)
+    assert(not set_instance_to_group.does_vm_exist_in_cfme(), "Instance exists")
+
+
+@pytest.mark.skipif('True')
```

Comment:
This shouldn't be in here, we should never just skip a test because with no reason